lib/prop/index.js

```diff
@@ -1,7 +1,8 @@
-import curry2 from '../_internal/_curry2'
-import view from '../view'
-import lensPath from '../lensPath'
+import _curry2 from '../_internal/_curry2'
 
-export default curry2(function prop(path, obj) {
-  return view(lensPath(path), obj)
+export default _curry2(function prop(p, obj) {
+  if (typeof p !== 'string' && typeof p !== 'number') {
+    throw new TypeError('Property should be either string or number')
```

Comment:
Do we need to throw error here? Ramda in any unexpected situation returns undefined

Reply:
hmmm